### ISSUE TITLE
[4.x]: Conditional Field is not displayed when rule depends on default dropdown value

### ISSUE THREAD

**Issue by vaelu** [craftcms/cms#12227]
### Description
When you have a field which only shows when a specific dropdown value is selected, and this dropdown value is the default value (which is already selected when you create a new entry) the field is not displayed. It does appear when you select another value and then select the initial/default value again.

### Steps to reproduce

1. Create a dropdown field with 2 or more values (for example, value A and value B).
2. Create another field. Add a condition to this field that it's only displayed when value A is selected in the dropdown field.
3. Create a new entry draft.

### Expected behavior
In the created entry draft, value A from the dropdown field is selected by default. Since this is the condition for displaying the other field, it should be visible.

### Actual behavior
Unfortunately, the field is not visible. Only when you select value B, and then select value A again, it will become visible.

### Workaround
Instead of creating a condition "Show when dropdown field has value A", create a condition which is "Show when dropdown field has not value B".

### Craft CMS version

4.31

### PHP version

8.1.11

### Operating system and version

Linux 5.15.68.1

### Database type and version

MySQL 10.4.26

**Comment by i-just** [comment]
Hi, thanks for reaching out. I can't replicate this issue. Do you have any other conditions for that entry or are you using any plugins?

**Comment by mmikkel** [comment]
I believe the issue here is that for new elements, Dropdown fields will technically have a `null` value – even if they are visually presented to authors as being set to the default value. Hence, any conditionals depending on the Dropdown field being set to the default value, will evaluate to `false` (and it's not apparent to the author why, because it looks like it's set to the default).

**Comment by vaelu** [comment]
@mmikkel Is there any reason for them to have `null` as value when the default value is saved? That does not sound like wanted behaviour to me.

**Comment by mmikkel** [comment]
Disregard my comment above; I'm actually not able to reproduce this anymore (on 4.3.1). I have seen it happen before though.

**Comment by janreges** [comment]
Hi,

I have related, maybe the same issue. I'm using latest Craft CMS 4.3.1.

I have in my entry type "Article" dropdown "Source type" with available values "Publication", "Audio", "Video". Also "Article" has "Audio URL" field with defined entry condition "Source type [is one of] Audio".

When I am adding new Article and I choose Source type = Audio, nothing happens. I expected that "Audio URL" will appear. 

I tried to use the entry condition for the first time today, so unfortunately I can't verify if it worked in a previous version.

**Comment by vaelu** [comment]
@janreges Just to verify: what value (not the label) does your dropdown have?

**Comment by janreges** [comment]
@vaelu - here is copied outer HTML of my source type element:

Btw, it doesn't work either in the add or edit form. At the same time, it does not work either in the case of defined/undefined default value of dropdown, or in case of any change to the selectbox.

In the Network console, I can see that all JS/CSS have been loaded correctly. At the same time, Console does not contain any errors.

```html
<select id="fields-sourceType" name="fields[sourceType]">
   <option value="base64:cHVibGljYXRpb24=">Publikace</option>
   <option value="base64:YXVkaW8=">Audio</option>
   <option value="base64:dmlkZW8=">Video</option>
</select>
```

I don't know why there are base64 values ... in dropdown field definition i have values `publication`, `audio` and `video`.

**Comment by janreges** [comment]
Also in entry conditions form there is select with original value `audio`. 

So I'm guessing that the javascript-based control of the conditional display doesn't work when Craft CMS uses base64 values in the dropdown for some reason?

```html
<select id="element-1e87036f-f97c-4ec7-9c2d-b4210d923899-elementCondition-conditionRules-1-multiselect" name="element-1e87036f-f97c-4ec7-9c2d-b4210d923899[elementCondition][conditionRules][1][values][]" multiple="multiple" style="display: none;" tabindex="-1" class="selectized">
   <option value="audio" selected="selected">Audio</option>
</select>
```

**Comment by vaelu** [comment]
@janreges Okay, and your values in the Craft backend's field settings? Are there also base64 values?

And no, the cause must be something else, because my values are `1`, `2` and `3`.

**Comment by janreges** [comment]
@vaelu - dropdown field settings contains this config (value -> label):

```
publication -> Publikace
audio -> Audio
video -> Video
```

**Comment by vaelu** [comment]
Update: I tested it again and noticed that it has to do with the "default value" setting. In my first test, I haven't set a default value. Now I set the first value as "default value" and it works.
![image](https://user-images.githubusercontent.com/72358141/199051580-406d0e47-c5a4-4230-99d2-13470a002294.png)

Can you check on your end and verify it? @janreges

**Comment by i-just** [comment]
@vaelu your last update explains why I wasn't able to replicate this - thanks for clarifying. When you initially said you have a rule that depends on a default dropdown value, I was explicitly checking the "default value" checkbox and testing against that in the 3 latest releases. With the updated description, I can replicate this behaviour - thanks!

@janreges let me just clarify if I get your problem correctly. You have `Source type` dropdown field with `publication`, `audio` and `video` values; you have a condition set on `Audio URL` field that says "Source type [is one of] Audio" (an you've applied that and saved your entry type) and then when you create or edit entry of that type you don't see the `Audio URL` field even if you switch to a different item in the dropdown and then swith to `audio` again?

**Comment by vaelu** [comment]
@i-just Yeah, sorry, it was a bit confusing because by "default value" I meant the first one. I edited in the initial issue.

**Comment by vaelu** [comment]
@brandonkelly Thanks for clarifying.

Just as a question, wouldn't it make more sense and improve UX when Craft would add an `<option>` which is disabled (for example "Select an option") when there is no default dropdown value set. The value of this option would be null, and it would not be savable when the field is marked as required.

**Comment by brandonkelly** [comment]
Something to consider. Do you mind opening up a [new discussion](https://github.com/craftcms/cms/discussions/new?category=ideas) with that idea?

**Comment by janreges** [comment]
For future readers. I would like to clarify the issues that I mentioned in this issue.

Thanks for consulting @i-just via Discord. My issue with conditional fields not working was caused by having `autosaveDrafts: false' set. If the automatic saving of drafts is turned off, even the UI response to conditional fields does not work at all. @i-just will add this information as a warning to the documentation of the `autosaveDrafts` setting.
